### DISCUSSION TITLE
Compatibility with write-protected source directories

### DISCUSSION THREAD

**Discussion by ropeladder** [quarto-dev/quarto-cli#7254] [Q&A] [answered]
### Description

There are a number of similar issues that touch on this functionality ( #1666, #6518 ) but I couldn't find any that address it explicitly.

We are trying to render pdfs from qmd source directories that are write protected. So far, here are the things I've found that stand in the way: (as of v1.3.450)

1. Automatic addition/creation of `.gitignore` (can be circumvented by adding "`/.quarto/`" to the repo's `.gitignore` file)
1. Creation of `.quarto/` directory in render project folder that contains the cache and ??? (`--no-cache` doesn't appear to avoid this entirely)
1. Temporary files: unclear whether there are roadblocks that remain here; there is an `--execute-dir` flag dir
1. Output: `--output-dir` flag now appears to work, but only with project-based workflows

**Comment by mcanouil** [comment, marked as answer]
Possibly address in the future with:
- #6518

In the meantime, there is an easy workaround/solution which is to not run Quarto inside the non writable environment:

```sh
cp -r my-quarto-project /tmp
quarto render /tmp/my-quarto-project
```

Note that the issue would be the same with Pandoc, rmarkdown, etc.
Executing something in a non-writable space is rarely working ok.
